src/test/rgw/test_redis_lock.cc

```diff
@@ -0,0 +1,175 @@
+
+        return_code = rgw::redislock::assert_locked(conn, name, cookie, yield);
+        ASSERT_EQ(return_code, 0);
+      },
+      [this](std::exception_ptr eptr) {
+        conn->cancel();
+        if (eptr) std::rethrow_exception(eptr);
+      });
+  io.run();
+}
+
+TEST_F(RGWRedisLockTest, Unlock) {
+  io.restart();
+  boost::asio::spawn(
+      io,
+      [this](boost::asio::yield_context yield) {
+        const std::string name = "lock:unlock";
+        const std::string cookie = "mycookie";
+        int duration = 12000;
+
+        int return_code =
+            rgw::redislock::lock(conn, name, cookie, duration, yield);
+        ASSERT_EQ(return_code, 0);
+
+        return_code = rgw::redislock::unlock(conn, name, cookie, yield);
+        ASSERT_EQ(return_code, 0);
+
+        return_code = rgw::redislock::assert_locked(conn, name, cookie, yield);
+        ASSERT_NE(return_code, 0);
+      },
+      [this](std::exception_ptr eptr) {
+        conn->cancel();
+        if (eptr) std::rethrow_exception(eptr);
+      });
+  io.run();
+}
+
+TEST_F(RGWRedisLockTest, Renew) {
+  io.restart();
+  boost::asio::spawn(
+      io,
+      [this](boost::asio::yield_context yield) {
+        const std::string name = "lock:renew";
+        const std::string cookie = "mycookie";
+        int duration = 12000;
+
+        int return_code =
+            rgw::redislock::lock(conn, name, cookie, duration, yield);
+        ASSERT_EQ(return_code, 0);
+
+        return_code = rgw::redislock::lock(conn, name, cookie, duration, yield);
+        ASSERT_EQ(return_code, 0);
+      },
+      [this](std::exception_ptr eptr) {
+        conn->cancel();
+        if (eptr) std::rethrow_exception(eptr);
+      });
+}
+
+TEST_F(RGWRedisLockTest, MultiLock) {
+  io.restart();
+  boost::asio::spawn(
+      io,
+      [this](boost::asio::yield_context yield) {
+        int duration = 12000;
+        const std::string name = "lock:multi";
+        const std::string clientCookie1 = "mycookie1";
+        const std::string clientCookie2 = "mycookie2";
+
+        int return_code =
+            rgw::redislock::lock(conn, name, clientCookie1, duration, yield);
+        ASSERT_EQ(return_code, 0);
+
+        return_code =
+            rgw::redislock::lock(conn, name, clientCookie2, duration, yield);
+        ASSERT_EQ(return_code, -EBUSY);
+      },
+      [this](std::exception_ptr eptr) {
+        conn->cancel();
+        if (eptr) std::rethrow_exception(eptr);
+      });
+  io.run();
+}
+
+TEST_F(RGWRedisLockTest, Timeout) {
+  io.restart();
+  boost::asio::spawn(
+      io,
+      [this](boost::asio::yield_context yield) {
+        int duration = 500;
+        const std::string name = "lock:timeout";
+        const std::string cookie = "mycookie";
+
+        int return_code =
+            rgw::redislock::lock(conn, name, cookie, duration, yield);
+        ASSERT_EQ(return_code, 0);
+
+        boost::asio::steady_timer timer(io, std::chrono::milliseconds(500));
```

Comment:
please use longer times, so that the test does not fail in the lab

Reply:
Changed the wait to 1000 ms ( 1 sec).
Redis TTL is 500 ms.